File: Fortran/randomSimplexMatrix.f
subroutine randomSimplexMatrix(n, m, y)
    implicit none
    integer, intent(in) :: n, m
    real(kind=8), dimension(m, n) :: y
    real(kind=8), dimension(m, n) :: C, p, xi, S
    integer :: i, j

    call random_number(C)
    C = C + tiny(1.0)  ! Ensure no zero values for log transformation
    C = 1.0 - C         ! Transform to open interval (0,1]
    
    do i = 1, m
        do j = 1, n
            p(i,j) = -log(C(i,j))
        end do
    end do
    
    do i = 1, m
        S(i) = sum(p(i,:))
        y(i,:) = p(i,:) / S(i)
    end do
    
    ! Check each row sums to approximately 1
    do i = 1, m
        if (abs(sum(y(i,:)) - 1.0) > 1.0e-10) then
            write(*,*) 'Error: Rows do not sum to 1.'
            stop
        end if
    end do
    
end subroutine randomSimplexMatrix

File: randomSimplexMatrix.m
function y = randomSimplexMatrix(n,m)
    
    % Generate m x n matrix of random numbers between 0 and 1
    C = rand(m, n); 
    
    % Transform elements of C to ensure each row sums to 1
    p = -log(C);
    S = sum(p, 2);

    % Sum each row to 1
    y = p ./ S;  
    
    % Test each row sums to 1
    row_sums = sum(y, 2);
    assert(all(abs(row_sums - 1) < 1e-10), 'Error: Rows do not sum to 1.');
end